### PR TITLE
Add docs and specs to String.Chars and List.Chars

### PR DESCRIPTION
**Pull request by whatyouhide**

### CODE REVIEW

File: lib/elixir/lib/list/chars.ex [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
   The only function required to be implemented is
-  `to_charlist` which does the conversion.
+  `to_charlist/1` which does the conversion.
 
   The `to_charlist/1` function automatically imported
   by `Kernel` invokes this protocol.
   """
 
+  @doc """
+  Converts `terms` to a charlist.
```
Comment (fishcakez) [inline lib/elixir/lib/list/chars.ex]: `term`

File: lib/elixir/lib/list/chars.ex [hunk 2]
```diff
@@ -1,14 +1,18 @@
 defprotocol List.Chars do
   @moduledoc ~S"""
-  The List.Chars protocol is responsible for
-  converting a structure to a list (only if applicable).
+  The `List.Chars` protocol is responsible for
+  converting a structure to a charlist (only if applicable).
```
Comment (josevalim) [inline lib/elixir/lib/list/chars.ex]: The first line should be a short summary.

File: lib/elixir/lib/string/chars.ex [hunk 3]
```diff
@@ -5,15 +5,19 @@ defprotocol String.Chars do
   The `String.Chars` protocol is responsible for
```
Comment (josevalim) [inline lib/elixir/lib/string/chars.ex]: The first line should be a short summary.

Comment (whatyouhide) [inline lib/elixir/lib/string/chars.ex]: Oh good catch, I only fixed minor things here and didn't notice.

**Pull request merged**